src/pages/Wallet/SendPage.tsx

```diff
@@ -17,22 +17,63 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
 import { Send } from 'lucide-react'
-import { useContext, useState } from 'react'
+import { ChangeEvent, useContext, useState } from 'react'
 import { useHistory } from 'react-router'
 import styled, { useTheme } from 'styled-components'
 
 import { GlobalContext } from '../../App'
 import { Button } from '../../components/Buttons'
+import ExpandableSection from '../../components/ExpandableSection'
 import InfoBox from '../../components/InfoBox'
 import Input from '../../components/Inputs/Input'
 import { ModalContext } from '../../components/Modal'
 import { Section } from '../../components/PageComponents/PageContainers'
 import Spinner from '../../components/Spinner'
 import { checkAddressValidity } from '../../utils/addresses'
 import { getHumanReadableError } from '../../utils/api'
-import { convertToQALPH } from '../../utils/numbers'
+import { MINIMAL_GAS_AMOUNT, MINIMAL_GAS_PRICE } from '../../utils/constants'
+import { abbreviateAmount, convertToQALPH } from '../../utils/numbers'
 import { WalletContext } from './WalletRootPage'
 
+const onAmountInputValueChange = ({
+  amount,
+  minAmount,
+  stateSetter,
+  errorMessage,
+  currentErrorState,
+  errorStateSetter,
+  shouldConvertToQALPH
+}: {
+  amount: string
+  minAmount: bigint
+  errorMessage: string
+  stateSetter: (v: string) => void
+  currentErrorState: string
+  errorStateSetter: (v: string) => void
+  shouldConvertToQALPH?: boolean
+}) => {
+  let amountNumber
+
+  try {
+    amountNumber = BigInt(shouldConvertToQALPH ? convertToQALPH(amount) : amount)
+  } catch (e) {
+    console.log(e)
+    return
+  }
+
+  if (amountNumber < minAmount) {
+    errorStateSetter(errorMessage)
+  } else {
+    if (currentErrorState) errorStateSetter('')
+  }
+
+  stateSetter(amount)
+}
```

Comment:
I see that you are trying to keep the code DRY with this function, but from my side, I don't see a lot of added value. I feel like we unnecessarily increase the number of lines and the complexity of the code. Even if it looks a bit less DRY, I would simplify the code and reduce the total number of lines of the file. Lemme know what you think.

![image](https://user-images.githubusercontent.com/1579899/148023890-be1153a9-34f9-49a2-a7f7-1fcef8769dc0.png)

Reply:
Mhh on this one I think that I prefer my version. The increased number of line is mostly caused by the declaration of the function's params and their typings, not by some logic that is hard to comprehend. 

I actually think that the condensed callbacks make it easier to understand the component logic & make it easier to identify and update the error messages if needed (readability can be improved further if we "hide" the `onAmountInputValueChange` at the bottom of the file.)

Reply:
Alright, agreed. Let's move the declaration of the `onAmountInputValueChange` utility to the end of file (before styling) so that it's not the first thing we read when opening the file (I find it nicer to read the component code directly and not implementation details).